tests/pytype_test.py

```diff
@@ -161,9 +161,8 @@ def get_missing_modules(files_to_test: Sequence[str]) -> Iterable[str]:
         for pkg in read_dependencies(distribution).external_pkgs:
             # See https://stackoverflow.com/a/54853084
             top_level_file = os.path.join(
-                # Fixed in #9747
-                pkg_resources.get_distribution(pkg).egg_info,  # type: ignore[attr-defined]  # pyright: ignore[reportGeneralTypeIssues]
-                "top_level.txt",
+                pkg_resources.get_distribution(pkg).egg_info,  # type: ignore[arg-type]  # pyright: ignore[reportGeneralTypeIssues]
+                "top_level.txt",  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
Something like this? The code already implicitely made this assert assumption.
```suggestion
            egg_info = pkg_resources.get_distribution(pkg).egg_info
            assert isinstance(egg_info, str)
            # See https://stackoverflow.com/a/54853084
            top_level_file = os.path.join(egg_info, "top_level.txt")
```

Reply:
Wanna make another PR? I'll happily close this one if you can get the CI green in a more principled way 😀

Reply:
There: https://github.com/python/typeshed/pull/9849#issue-1611960523 Feel free to suggest changes if you want a different assert (truthy, is not None, ...) or a message with said assert "Missing egg_info on package ..."